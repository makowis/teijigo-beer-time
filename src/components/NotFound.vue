<template>
  <div class="not-found">
    <h1>404 NOT FOUND</h1>
    <p>このページは存在しません。</p>
    <img src="../assets/nomikai_salaryman.png" alt="飲み会のイラストです。ビール飲みましょう">

  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'not-found',
});
</script>

<style scoped>
</style>
