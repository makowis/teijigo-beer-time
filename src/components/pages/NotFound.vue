<template>
  <tbt-default>
    <div class="not-found">
      <tbt-h1>404 NOT FOUND</tbt-h1>
      <p>このページは存在しません。</p>
      <img
        src="@/assets/nomikai_salaryman.png"
        alt="飲み会のイラストです。ビール飲みましょう"
      />
    </div>
  </tbt-default>
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import TbtDefault from '@/components/templates/TbtDefault.vue';
import TbtH1 from '@/components/atoms/TbtH1.vue';

@Component({
  components: {
    TbtDefault,
    TbtH1,
  },
})
export default class NotFound extends Vue {}
</script>

<style scoped></style>
