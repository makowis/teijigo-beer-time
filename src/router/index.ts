import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/pages/Top.vue';
import Disco from '@/components/pages/Disco.vue';
import Bbs from '@/components/pages/Bbs.vue';
import Member from '@/components/pages/Member.vue';
import Beenya from '@/components/pages/Beenya.vue';
import NotFound from '@/components/pages/NotFound.vue';

Vue.use(Router);

export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
    {
      path: '/bbs',
      name: 'Bbs',
      component: Bbs,
    },
    {
      path: '/member',
      name: 'Member',
      component: Member,
    },
    {
      path: '/beenya',
      name: 'Beenya',
      component: Beenya,
    },
    {
      path: '*',
      name: 'NotFound',
      component: NotFound,
    },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (savedPosition) {
      return savedPosition;
    }
    return { x: 0, y: 0 };
  },
});
