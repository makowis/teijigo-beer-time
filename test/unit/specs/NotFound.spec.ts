import Vue from 'vue';
import NotFound from '@/components/pages/NotFound.vue';
import router from '@/router/index';

describe('NotFound.vue', () => {
  const Constructor = Vue.extend(NotFound);
  const vm = new Constructor({ router }).$mount();

  it('should match the snapshot', () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
