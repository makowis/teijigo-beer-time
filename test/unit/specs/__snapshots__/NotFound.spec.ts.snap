// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound.vue should match the snapshot 1`] = `
<div
  id="app"
>
  <nav
    class="menuContainer"
    id="menu-container"
  >
    <ul
      class="menu"
    >
      <li
        class="menuItem"
      >
        <a
          class="router-link-exact-active router-link-active"
          href="/"
        >
           トップ 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/disco"
        >
           DISCOGRAPHY 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/member"
        >
           メンバー紹介 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/bbs"
        >
           一言掲示板 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/beenya"
        >
           びーにゃの部屋 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          href="https://teijigo.booth.pm/items/1093403"
          target="_blank"
        >
          
        通販(BOOTH)
      
        </a>
      </li>
    </ul>
  </nav>
   
  <div
    class="mainContainer"
    id="main-container"
  >
    <a
      class="router-link-exact-active router-link-active"
      href="/"
    >
      <img
        class="logo"
        src="@/assets/logo_1.png"
      />
    </a>
     
    <div
      class="not-found"
    >
      <h1
        class="h1"
      >
        404 NOT FOUND
      </h1>
       
      <p>
        このページは存在しません。
      </p>
       
      <img
        alt="飲み会のイラストです。ビール飲みましょう"
        src="@/assets/nomikai_salaryman.png"
      />
    </div>
     
    <footer
      class="footer"
    >
      <p>
        ©定時後ビールタイム
      </p>
    </footer>
  </div>
</div>
`;
